.github/workflows/ci.yml

```diff
@@ -15,10 +15,27 @@ on:
 
 jobs:
 
-  CI:
-    uses: ./.github/workflows/merge-gate.yml
-    secrets: inherit
+  checks:
+    name: Checks
+    uses: ./.github/workflows/checks.yml
+
+  Tests:
+    name: Tests
+    uses: ./.github/workflows/tests_with_coverage.yml
+
+  # This job ensures inputs have been executed successfully.
+  approve-merge:
+    name: Allow Merge
+    runs-on: ubuntu-latest
+    # If you need additional jobs to be part of the merge gate, add them below
+    needs: [ Tests ]
+
+    # Each job requires a step, so we added this dummy step.
+    steps:
+      - name: Approve
+        run: |
+          echo "Merge Approved"
 
   Metrics:
-    needs: [ CI ]
+    needs: [ Tests ]
```

Comment:
```suggestion
    needs: [ checks, tests ]
```

Reply:
Did the second option from comment above